internal/encryption/interfaces.go

```diff
@@ -0,0 +1,11 @@
+// Copyright 2024 Canonical Ltd.
+// SPDX-License-Identifier: AGPL-3.0
+
+package encryption
```

Comment:
object in encryption seems to be used only in the authentication package, wouldn't it be better off under that instead of having a package on its own?

not a biggie, just don't wanna go nuclear with the amount of mini packages we have

Reply:
Not a huge fan of mixing packages but I see what you mean, the language forces us already to have too many packages.
I'll merge it